src/ZoneServer/World/Actors/Characters/Character.cs

```diff
@@ -528,6 +529,9 @@ public void Warp(int mapId, Position pos)
 			if (!ZoneServer.Instance.World.TryGetMap(mapId, out var map))
 				throw new ArgumentException($"Map with id '{mapId}' doesn't exist in world.");
 
+			if (ZoneServer.Instance.Conf.World.BlueOrbFollowWarp || ZoneServer.Instance.Conf.World.BlueOrbPetSystem)
```

Comment:
I'm not quite happy with this code duplication. Doesn't it feel more like there should be an event one could subscribe to to be informed that a player warps? That way we could even decouple this from the character, which arguably shouldn't have to care about whether it even has a follower.

Reply:
This is the one thing I haven't touched.  I think the current implementation is still fairly simple and it's still a big improvement over having the pet constantly check to see if the player left, which caused a number of small issues.  Maybe adding an event could be a future enhancement.